app/models/manageiq/providers/openstack/cloud_manager/refresher.rb

```diff
@@ -68,6 +68,13 @@ def preprocess_targets
           end
         end
 
+        # if the cloud manager is due to be refreshed, make sure it's the first target.
+        cloud_manager_index = all_targets.find_index { |t| t.kind_of? ManageIQ::Providers::Openstack::CloudManager }
```

Comment:
isn't this stored as 
```ruby
@targets_by_ems_id {
  1 => [cloud_ems, target1, target2] ,
  2 => [network_ems],
  3 => [cinder_ems],
  4 => [swift_ems]
```

then the sorting like this would not help I think

Reply:
hrm, could be. I'll recheck.

Reply:
Looks like you're right, but in that case there's no sorting that could help here. `ems_refresher_mixin.rb:refresh` would have to do the sorting, and adding a special case there seems bad.

Reply:
So, it is pretty horrible, but you might be able to change the @targets_by_ems_id here. Probably before returning it, you would @targets_by_ems_id = {}, and then write it there in a new order. There is @ems_by_ems_id that you can use, with combination of keys of targets_by_ems_id, s you don;t have to query for EMS.